src/MyFrame.cpp

```diff
@@ -2025,6 +2038,35 @@ MyFrame::OnRevealLetter(wxCommandEvent & WXUNUSED(evt))
 }
 
 void
+MyFrame::OnConvertToNormal(wxCommandEvent & WXUNUSED(evt))
+{
+    wxASSERT(m_puz.IsDiagramless());
+
+    if (! XWordPrompt(this, _T("This will convert this diagramless puzzle to a normal puzzle, ")
+                            _T("revealing all black squares and overwriting any letters entered ")
+                            _T("where black squares should be.  Continue?")))
+        return;
+
+    m_puz.ConvertDiagramlessToNormal();
+
+    // Turn off the diagramless menu items
+    EnableDiagramless(false);
+
+    // Ensure focus moves to a different word as it may be on a black square. We move twice to
+    // ensure that the old word is cleared, as it may not have the correct boundaries.
+    m_XGridCtrl->SetFocusedSquare(m_XGridCtrl->LastWhite(), puz::ACROSS);
+    m_XGridCtrl->SetFocusedSquare(m_XGridCtrl->FirstWhite(), puz::ACROSS);
+
+    // Refresh the grid UI
+    m_XGridCtrl->Refresh();
+
+    // Send puzzle updated event
+    wxPuzEvent evt(wxEVT_PUZ_LETTER, GetId());
+    evt.SetSquare(m_XGridCtrl->GetFocusedSquare());
+    GetEventHandler()->ProcessEvent(evt);

```

Comment:
You should be able to simplify this whole bit to m_XGridCtrl->SendEvent(wxEVT_PUZ_LETTER) . . . I'm seeing a similar block of code in MyFrame::OnEraseGrid, which I'm guessing was written before I added that function to XGridCtrl, so that's obviously my fault for not fixing OnEraseGrid